### PR TITLE
Adds parseJson builtin method

### PR DESCRIPTION
**Pull request by simonswine**
This adds an implementation of `std.parseJson`

For my use case I need a way to parse a string float in jsonnet.

https://github.com/google/jsonnet/blob/a8769d48f96e0d419cd18190f16c294cce134f4d/core/vm.cpp#L1579

### CODE REVIEW

File: crates/jrsonnet-evaluator/src/integrations/serde.rs [hunk 1]
```diff
@@ -46,6 +46,12 @@ impl TryFrom<&Val> for Value {
 	}
 }
 
+impl From<serde_json::Error> for LocError {
+	fn from(e: serde_json::Error) -> Self {
+		LocError::new(RuntimeError(format!("Error parsing JSON {:?}", e).into()))
```
Comment (CertainLach) [inline crates/jrsonnet-evaluator/src/integrations/serde.rs:51]: Please use `Error` variant for errors (I have some plans for it)

Comment (CertainLach) [inline crates/jrsonnet-evaluator/src/integrations/serde.rs:51]: Please use `Error` variant for errors (I have some plans for this enum)

Comment (CertainLach) [inline crates/jrsonnet-evaluator/src/integrations/serde.rs:51]: Please use `Error` variant for errors (I have some plans for it)

**Review by CertainLach — CHANGES_REQUESTED** [review]

File: crates/jrsonnet-evaluator/src/lib.rs [hunk 2]
```diff
@@ -801,6 +801,21 @@ pub mod tests {
...
+			r#"std.parseJson('{"a": -1,"b": 1,"c": 3.141,"d": []}')"#,
+			r#"{"a": -1,"b": 1,"c": 3.141,"d": []}"#
+		);
+		// TODO: this should in fact fail as is no proper JSON syntax
+		assert_json!(
+			r#"std.parseJson("{a:-1, b:1, c:3.141, d:[]}")"#,
+			r#"{"a": -1,"b": 1,"c": 3.141,"d": []}"#
+		);
+		// TODO: this is also no valid JSON
+		assert_json!(r#"std.parseJson('local x = 2; x * x')"#, r#"4"#);
```
Comment (simonswine) [inline crates/jrsonnet-evaluator/src/lib.rs:816]: Not too sure how bad it is that you now can actually execute jsonnet within `std.parseJson()`

File: crates/jrsonnet-evaluator/src/integrations/serde.rs (hunk 1 above)
Comment (simonswine) [reply crates/jrsonnet-evaluator/src/integrations/serde.rs]: Not entirely sure what that meant, but I guess by using `state.evaluate_snippet_raw` that should no longer be a problem

File: crates/jrsonnet-evaluator/src/builtin/mod.rs [hunk 3]
```diff
@@ -164,6 +166,19 @@ fn builtin_object_has_ex(
 	})
 }
 
+fn builtin_parse_json(
+	context: Context,
+	_loc: Option<&ExprLocation>,
+	args: &ArgsDesc,
+) -> Result<Val> {
+	parse_args!(context, "objectFieldsEx", args, 2, [
```
[the remark this answers is not in the document]
Comment (simonswine) [reply crates/jrsonnet-evaluator/src/builtin/mod.rs]: Sorry about that, was just an oversight after copy-pasting

File: crates/jrsonnet-evaluator/src/builtin/mod.rs [hunk 4]
```diff
@@ -8,6 +8,7 @@ use format::{format_arr, format_obj};
 use jrsonnet_interner::IStr;
 use jrsonnet_parser::{ArgsDesc, BinaryOpType, ExprLocation};
 use jrsonnet_types::ty;
+use serde_json::Value;
```
[the remark this answers is not in the document]
Comment (simonswine) [reply crates/jrsonnet-evaluator/src/builtin/mod.rs]: Very good point, I think that makes a lot more sense. I guess the main difference now is that `std.parseJson` is not partically strict and allows execution of any Jsonnet valid syntax. Any ways to make it stricter?

**Comment by simonswine** [comment]
Thank you for starting this project and for your quick review. 

You will notice this is one of my first rust contributions, so it might take a bit longer to get it all in shape :slightly_smiling_face:

File: crates/jrsonnet-evaluator/src/builtin/mod.rs (hunk 4 above)
[the remark this answers is not in the document]
Comment (CertainLach) [reply crates/jrsonnet-evaluator/src/builtin/mod.rs]: > Any ways to make it stricter?
Not yet, but i think it is acceptable for now, as `std.parseJson` itself is very niche method :D

**Comment by CertainLach** [comment]
Thanks!

**Pull request merged**